service/expression.js

```diff
@@ -15,12 +17,13 @@ class ExpressionService {
     return expression;
   }
 
-  async fetchExpressions(userId) {
-    const result = await expressionRepository.getExpressionsByUser(userId);
+  async fetchExpressions(userId, role) {
+    const result =
+      role === regularRole
+        ? await userRepository.getUserExpressions(userId)
```

Comment:
Why are you getting `results` from two different repos based on the role. In that results you have expressions so they should come from the expressions repo.